src/components/inputs/date/date.component.tsx

```diff
@@ -72,7 +72,7 @@ const DateField: React.FC<FormFieldInputProps> = ({ field, value: dateValue, err
     [field.datePickerFormat, field.label, t],
   );
 
-  return sessionMode == 'view' || sessionMode == 'embedded-view' ? (
+  return sessionMode == 'view' || sessionMode == 'embedded-view' || isTrue(field.readonly) ? (
```

Comment:
Shouldn't [this](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/fea4fb83eb1bad938bcad8f15c04a66543df61ad/src/components/inputs/date/date.component.tsx#L118) and [this](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/fea4fb83eb1bad938bcad8f15c04a66543df61ad/src/components/inputs/date/date.component.tsx#L121) suffice?

Reply:
We're currently using the OpenMRS date picker for the date field (calendar) and time picker if specified. Providing the readonly works as intended for the time picker but this doesn't propagate to the OpenMRS date picker as seen in this screenshot. So the fall back for that was to use the view only rendering hence the need for this. 

<img width="1649" alt="Screenshot 2024-09-26 at 23 12 29" src="https://github.com/user-attachments/assets/6e956b53-c965-4e08-a787-927c308636d6">